crates/ruff_server/src/lint.rs

```diff
@@ -70,16 +69,16 @@ pub(crate) fn check(
         &linter_settings.namespace_packages,
     );
 
-    let source_type = PySourceType::default();
-
-    // TODO(jane): Support Jupyter Notebooks
-    let source_kind = SourceKind::Python(contents.to_string());
+    let source_type = query.source_type();
 
     // Tokenize once.
-    let tokens: Vec<LexResult> = ruff_python_parser::tokenize(contents, source_type.as_mode());
+    let tokens: Vec<LexResult> =
+        ruff_python_parser::tokenize(source_kind.source_code(), source_type.as_mode());
+
+    let index = LineIndex::from_source_text(source_kind.source_code());
```

Comment:
Can we separate the logic for Notebook and text document to avoid building the index multiple times? But, I see that you want to get remove the index altogether for text document so maybe this isn't worth looking into right now.

Reply:
We could separate the logic but I don't think it's worth the technical investment IMO if we plan to eventually remove this anyway.